src/System.Runtime/tests/Performance/Perf.String.cs

```diff
@@ -0,0 +1,243 @@
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void StartsWith(int size)
+        {
+            string testString = PerfUtils.CreateString(size);
+            string subString = testString.Substring(0, testString.Length / 4);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.StartsWith(subString);
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void Empty(int size)
+        {
+            string result;
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                {
+                    result = string.Empty; result = string.Empty; result = string.Empty;
+                    result = string.Empty; result = string.Empty; result = string.Empty;
+                    result = string.Empty; result = string.Empty; result = string.Empty;
+                }
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void Substring_int(int size)
+        {
+            int startIndex = size / 2;
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.Substring(startIndex);
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void Substring_int_int(int size)
+        {
+            int startIndex = size / 2;
+            int length = size / 4;
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.Substring(startIndex, length);
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void ToLower(int size)
+        {
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.ToLower();
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void ToUpper(int size)
+        {
+            string testString = PerfUtils.CreateString(size);
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.ToUpper();
+        }
+
+        [Benchmark]
+        [MemberData("TestStringSizes")]
+        public void Trim(int size)
+        {
+            string testString = "   " + PerfUtils.CreateString(size) + "   ";
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    testString.Trim();
```

Comment:
In general, we should consider having some perf tests that exist to "lock in" optimizations we have in place.  For example, Trim is optimized to return the same string instance if nothing needs to be trimmed.